x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/string/HashTests.java

```diff
@@ -0,0 +1,132 @@
+import java.util.ArrayList;
+import java.util.HexFormat;
+import java.util.List;
+import java.util.function.Supplier;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+import static org.hamcrest.Matchers.startsWith;
+
+public class HashTests extends AbstractScalarFunctionTestCase {
+
+    public HashTests(@Name("TestCase") Supplier<TestCaseSupplier.TestCase> testCaseSupplier) {
+        this.testCase = testCaseSupplier.get();
+    }
+
+    @ParametersFactory
+    public static Iterable<Object[]> parameters() {
+        List<TestCaseSupplier> cases = new ArrayList<>();
+        for (String algorithm : List.of("MD5", "SHA", "SHA-224", "SHA-256", "SHA-384", "SHA-512")) {
+            cases.addAll(createTestCases(algorithm));
+        }
+        cases.add(new TestCaseSupplier("Invalid algorithm", List.of(DataType.KEYWORD, DataType.KEYWORD), () -> {
+            var input = randomAlphaOfLength(10);
+            return new TestCaseSupplier.TestCase(
+                List.of(
+                    new TestCaseSupplier.TypedData(new BytesRef("invalid"), DataType.KEYWORD, "algorithm"),
+                    new TestCaseSupplier.TypedData(new BytesRef(input), DataType.KEYWORD, "input")
+                ),
+                "HashEvaluator[algorithm=Attribute[channel=0], input=Attribute[channel=1]]",
+                DataType.KEYWORD,
+                is(nullValue())
+            ).withWarning("Line -1:-1: evaluation of [] failed, treating result as null. Only first 20 failures recorded.")
+                .withWarning("Line -1:-1: java.security.NoSuchAlgorithmException: invalid MessageDigest not available")
+                .withFoldingException(InvalidArgumentException.class, "invalid algorithm for []: invalid MessageDigest not available");
+        }));
+        return parameterSuppliersFromTypedDataWithDefaultChecks(true, cases, (v, p) -> "string");
+    }
+
+    private static List<TestCaseSupplier> createTestCases(String algorithm) {
+        return List.of(
+            createTestCase(algorithm, DataType.KEYWORD, DataType.KEYWORD),
+            createTestCase(algorithm, DataType.KEYWORD, DataType.TEXT),
+            createTestCase(algorithm, DataType.TEXT, DataType.KEYWORD),
+            createTestCase(algorithm, DataType.TEXT, DataType.TEXT),
+            createLiteralTestCase(algorithm, DataType.KEYWORD, DataType.KEYWORD),
+            createLiteralTestCase(algorithm, DataType.KEYWORD, DataType.TEXT),
+            createLiteralTestCase(algorithm, DataType.TEXT, DataType.KEYWORD),
+            createLiteralTestCase(algorithm, DataType.TEXT, DataType.KEYWORD)
+        );
+    }
+
+    private static TestCaseSupplier createTestCase(String algorithm, DataType algorithmType, DataType inputType) {
+        return new TestCaseSupplier(algorithm, List.of(algorithmType, inputType), () -> {
+            var input = randomAlphaOfLength(10);
```

Comment:
Let's randomize the length of this one. Like @bpintea mentioned, it'd be nice to get empty string sometimes. Maybe:
```
rarely() ? "" : randomRealisticUnicodeOfLengthBetween(1, 1000)
```
It'll probably work fine, but more paranoia seems good.